src/binary_cache.cpp

```diff
@@ -59,12 +60,17 @@ static boost::filesystem::path ComputeUserCachePath()
 {
 #ifdef MIOPEN_CACHE_DIR
     std::string cache_dir = MIOPEN_CACHE_DIR;
+    std::string version;
 
-    std::string version =
+    version =
         std::to_string(MIOPEN_VERSION_MAJOR) + "." + std::to_string(MIOPEN_VERSION_MINOR) + "." +
         std::to_string(MIOPEN_VERSION_PATCH) + "." + MIOPEN_STRINGIZE(MIOPEN_VERSION_TWEAK);
-
     auto p = boost::filesystem::path{miopen::ExpandUser(cache_dir)} / version;
+
+    const char* const custom = miopen::GetStringEnv(MIOPEN_CUSTOM_CACHE_DIR{});
+    if(custom != nullptr && strlen(custom) > 0)
+        p = boost::filesystem::path{miopen::ExpandUser(custom)};
+
```

Comment:
These changes should be done in `ComputeUserCachePath`. Sys cache path is readonly ?

Reply:
ahmm...they are?

Reply:
You are right, my bad.